PropertyAccessException can occur at scheduler start

This is following recent work on database job removal

```
org.ow2.proactive.db.DatabaseManagerException: org.hibernate.PropertyAccessException: Null value was assigned to a property of primitive type setter of org.ow2.proactive.scheduler.core.db.JobData.scheduledTimeForRemoval
        at org.ow2.proactive.db.TransactionHelper.tryExecuteTransactionLoop(TransactionHelper.java:111)
        at org.ow2.proactive.db.TransactionHelper.executeReadOnlyTransaction(TransactionHelper.java:60)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.executeReadOnlyTransaction(SchedulerDBManager.java:1736)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.loadJobs(SchedulerDBManager.java:830)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.loadFinishedJobs(SchedulerDBManager.java:826)
        at org.ow2.proactive.scheduler.core.db.SchedulerStateRecoverHelper.recover(SchedulerStateRecoverHelper.java:133)
        at org.ow2.proactive.scheduler.core.SchedulerFrontend.initActivity(SchedulerFrontend.java:298)
        at org.objectweb.proactive.core.body.ActiveBody.run(ActiveBody.java:155)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.hibernate.PropertyAccessException: Null value was assigned to a property of primitive type setter of org.ow2.proactive.scheduler.core.db.JobData.scheduledTimeForRemoval
        at org.hibernate.property.access.spi.SetterMethodImpl.set(SetterMethodImpl.java:87)
        at org.hibernate.tuple.entity.AbstractEntityTuplizer.setPropertyValues(AbstractEntityTuplizer.java:611)
        at org.hibernate.tuple.entity.PojoEntityTuplizer.setPropertyValues(PojoEntityTuplizer.java:207)
        at org.hibernate.persister.entity.AbstractEntityPersister.setPropertyValues(AbstractEntityPersister.java:4605)
        at org.hibernate.engine.internal.TwoPhaseLoad.doInitializeEntity(TwoPhaseLoad.java:169)
        at org.hibernate.engine.internal.TwoPhaseLoad.initializeEntity(TwoPhaseLoad.java:125)
        at org.hibernate.loader.Loader.initializeEntitiesAndCollections(Loader.java:1132)
        at org.hibernate.loader.Loader.processResultSet(Loader.java:992)
        at org.hibernate.loader.Loader.doQuery(Loader.java:930)
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:336)
        at org.hibernate.loader.Loader.doList(Loader.java:2610)
        at org.hibernate.loader.Loader.doList(Loader.java:2593)
        at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2422)
        at org.hibernate.loader.Loader.list(Loader.java:2417)
        at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:501)
        at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:371)
        at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:216)
        at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1339)
        at org.hibernate.internal.QueryImpl.list(QueryImpl.java:87)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.batchLoadJobs(SchedulerDBManager.java:944)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.loadInternalJobs(SchedulerDBManager.java:930)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.access$400(SchedulerDBManager.java:109)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager$22.doInTransaction(SchedulerDBManager.java:850)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager$22.doInTransaction(SchedulerDBManager.java:830)
        at org.ow2.proactive.db.TransactionHelper.tryExecuteTransaction(TransactionHelper.java:122)
        at org.ow2.proactive.db.TransactionHelper.tryExecuteTransactionLoop(TransactionHelper.java:96)
        ... 8 more
Caused by: java.lang.IllegalArgumentException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.hibernate.property.access.spi.SetterMethodImpl.set(SetterMethodImpl.java:44)
        ... 33 more
```

--- a/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/JobInfoImpl.java
+++ b/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/JobInfoImpl.java
@@ -26,7 +26,6 @@
 package org.ow2.proactive.scheduler.rest.data;
 
 import java.util.Map;
-
 import org.ow2.proactive.scheduler.common.job.JobId;
 import org.ow2.proactive.scheduler.common.job.JobInfo;
 import org.ow2.proactive.scheduler.common.job.JobPriority;
@@ -68,6 +67,8 @@ public class JobInfoImpl implements JobInfo {
     private JobPriority jobPriority = JobPriority.NORMAL;
 
     private JobStatus jobStatus = JobStatus.PENDING;
+
+    private long scheduledTimeForRemoval;
 
     private boolean toBeRemoved;
 
@@ -233,6 +234,15 @@ public class JobInfoImpl implements JobInfo {
 
     public void setToBeRemoved() {
         this.toBeRemoved = true;
+    }
+
+    @Override
+    public long getScheduledTimeForRemoval() {
+        return scheduledTimeForRemoval;
+    }
+
+    public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
+        this.scheduledTimeForRemoval = scheduledTimeForRemoval;
     }
 
     @Override

--- a/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/JobInfoImpl.java
+++ b/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/JobInfoImpl.java
@@ -26,6 +26,7 @@
 package org.ow2.proactive.scheduler.rest.data;
 
 import java.util.Map;
+
 import org.ow2.proactive.scheduler.common.job.JobId;
 import org.ow2.proactive.scheduler.common.job.JobInfo;
 import org.ow2.proactive.scheduler.common.job.JobPriority;

--- a/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobInfo.java
+++ b/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobInfo.java
@@ -27,6 +27,7 @@ package org.ow2.proactive.scheduler.common.job;
 
 import java.io.Serializable;
 import java.util.Map;
+
 import org.objectweb.proactive.annotation.PublicAPI;
 
 

--- a/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobInfo.java
+++ b/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobInfo.java
@@ -27,7 +27,6 @@ package org.ow2.proactive.scheduler.common.job;
 
 import java.io.Serializable;
 import java.util.Map;
-
 import org.objectweb.proactive.annotation.PublicAPI;
 
 
@@ -190,6 +189,14 @@ public interface JobInfo extends Serializable {
      * is marked for removal or not.
      */
     boolean isToBeRemoved();
+
+    /**
+     * Returns the scheduled time for removal.
+     * If none is set, returns 0.
+     *
+     * @return the time at which the Job is to be removed.
+     */
+    long getScheduledTimeForRemoval();
 
     /**
      * Returns the generic information Map

--- a/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/job/JobInfoImpl.java
+++ b/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/job/JobInfoImpl.java
@@ -83,6 +83,10 @@ public class JobInfoImpl implements JobInfo {
     //DEFAULT MUST BE -1
     private long removedTime = -1;
 
+    /** job scheduled time for removal (for the housekeeping) */
+    // DEFAULT MUST BE 0 IF NOT SCHEDULED FOR REMOVAL
+    private long scheduledTimeForRemoval = 0;
+
     /** last updated time on the job*/
     private long lastUpdatedTime = -1;
 
@@ -410,6 +414,14 @@ public class JobInfoImpl implements JobInfo {
      */
     public Set<TaskId> getTasksSkipped() {
         return this.tasksSkipped;
+    }
+
+    public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
+        this.scheduledTimeForRemoval = scheduledTimeForRemoval;
+    }
+
+    public long getScheduledTimeForRemoval() {
+        return scheduledTimeForRemoval;
     }
 
     /*

--- a/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/job/JobInfoImpl.java
+++ b/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/job/JobInfoImpl.java
@@ -420,6 +420,7 @@ public class JobInfoImpl implements JobInfo {
         this.scheduledTimeForRemoval = scheduledTimeForRemoval;
     }
 
+    @Override
     public long getScheduledTimeForRemoval() {
         return scheduledTimeForRemoval;
     }

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java
@@ -249,7 +249,7 @@ public class JobData implements Serializable {
         internalJob.setUserSpace(getGlobalSpace());
         internalJob.setMaxNumberOfExecution(getMaxNumberOfExecution());
         internalJob.setOnTaskError(OnTaskError.getInstance(this.onTaskErrorString));
-
+        internalJob.setScheduledTimeForRemoval(getScheduledTimeForRemoval());
         return internalJob;
     }
 
@@ -263,6 +263,7 @@ public class JobData implements Serializable {
         jobRuntimeData.setInErrorTime(job.getInErrorTime());
         jobRuntimeData.setFinishedTime(job.getFinishedTime());
         jobRuntimeData.setRemovedTime(job.getRemovedTime());
+        jobRuntimeData.setScheduledTimeForRemoval(job.getScheduledTimeForRemoval());
         jobRuntimeData.setJobName(job.getName());
         jobRuntimeData.setDescription(job.getDescription());
         jobRuntimeData.setProjectName(job.getProjectName());
@@ -461,7 +462,7 @@ public class JobData implements Serializable {
         this.removedTime = removedTime;
     }
 
-    @Column(name = "SCHEDULED_TIME_FOR_REMOVAL")
+    @Column(name = "SCHEDULED_TIME_FOR_REMOVAL", nullable = false)
     public long getScheduledTimeForRemoval() {
         return scheduledTimeForRemoval;
     }

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java
@@ -1262,4 +1262,12 @@ public abstract class InternalJob extends JobState {
         this.taskFlowJob = taskFlowJob;
     }
 
+    public long getScheduledTimeForRemoval() {
+        return this.jobInfo.getScheduledTimeForRemoval();
+    }
+
+    public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
+        this.jobInfo.setScheduledTimeForRemoval(scheduledTimeForRemoval);
+    }
+
 }

